.travis.yml

```diff
@@ -33,15 +33,22 @@ matrix:
 #            - BREW_INSTALL=python3
 
 install:
-  - pip install tox
+  - pip install flake8 tox
 #  - |
 #    if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
 #      if [[ -n "$BREW_INSTALL" ]]; then
 #        brew update
 #        brew install "$BREW_INSTALL"
 #      fi
 #    fi
-#    pip install tox
+
+before_script:
+  # stop the build if there are Python syntax errors or undefined names
+  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+  if [[ $TOXENV == py37 ]]; then
+    flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics ;
```

Comment:
Adding a check for these "*showstopper*" issues is a nice improvement.  However, I would prefer if we only check the actual deliverable ``cmd2`` source code in the **cmd2** directory.  

As you pointed out, if we also apply these checks to code in the **examples** or **tests** directories, we will run into issues with ``app`` and ``self`` being undefined names.  This is a side-effect of the somewhat unconventional way in which ``cmd2`` supports Python scripting within ``cmd2`` applications.  These Python scripts which use ``app`` and ``self`` are ONLY intended to ever be run inside ``cmd2``'s scripting engine and not within a generic Python interpreter.

@cclauss What do you think?

@kmvanbrunt @kotfu @anselor Do any of you have any bright ideas how to make a small modification to the ``cmd2`` code so that ``flake8`` doesn't warn about ``app`` and/or ``self`` being undefined names in our example scripts and tests?